pkg/driver/node.go

```diff
@@ -861,8 +883,8 @@ func removeNotReadyTaint(k8sClient cloud.KubernetesAPIClient) error {
 
 	clientset, err := k8sClient()
 	if err != nil {
-		klog.V(4).InfoS("Failed to communicate with k8s API, skipping taint removal")
-		return nil //lint:ignore nilerr Failing to communicate with k8s API is a soft failure
+		klog.V(4).InfoS("Failed to setup k8s client")
+		return nil //lint:ignore nilerr If there are no k8s credentials, treat that as a soft failure
```

Comment:
Thanks for the quick turnaround on this - /lgtm

Q: Are there failure modes here where it would be desirable to retry here as opposed to returning nil?

Reply:
All that constructing the client does is try to load k8s credentials (it doesn't even reach out to the API server). Thus, all failures here are permanent failures, the most typical failure would be that no `kubeconfig` could be found.